app/models/authentication.rb

```diff
@@ -0,0 +1,19 @@
+class Authentication < ApplicationRecord
+  acts_as_tenant(:tenant)
+
+  has_one :encryption, :dependent => :destroy
+
+  validates :encryption, :presence => true
+
+  def password=(password)
```

Comment:
@syncrou should we call this secret or protected_data since we would be storing
 * passwords
 * OAuth tokens
 * HMAC secrets in this field.